cleanlab/datalab/datalab.py

```diff
@@ -0,0 +1,632 @@
+
+        Returns
+        -------
+        specific_issues :
+            A DataFrame containing the issues data for the specified issue type, or the entire `issues`
+            DataFrame if `issue_name` is `None`.
+
+            Additional columns are added to the DataFrame if `issue_name` is "label".
+        """
+        if issue_name is None:
+            return self.issues
+
+        specific_issues = self._get_matching_issue_columns(issue_name)
+        info = self.get_info(issue_name=issue_name)
+        if issue_name == "label":
+            specific_issues = specific_issues.assign(
+                given_label=info["given_label"], predicted_label=info["predicted_label"]
+            )
+
+        if issue_name == "outlier":
+            column_dict = {
+                k: info.get(k)
+                for k in ["nearest_neighbor", "distance_to_nearest_neighbor"]
+                if info.get(k) is not None
+            }
+            specific_issues = specific_issues.assign(**column_dict)
+
+        if issue_name == "near_duplicate":
+            column_dict = {
+                k: info.get(k)
+                for k in ["near_duplicate_sets", "distance_to_nearest_neighbor"]
+                if info.get(k) is not None
+            }
+            specific_issues = specific_issues.assign(**column_dict)
+        return specific_issues
+
+    def _get_matching_issue_columns(self, issue_name: str) -> pd.DataFrame:
+        columns = [col for col in self.issues.columns if issue_name in col]
+        if not columns:
+            raise ValueError(f"No columns found for issue type '{issue_name}'.")
+        return self.issues[columns]
+
+    def get_summary(self, issue_name: Optional[str] = None) -> pd.DataFrame:
+        """Summarize the issues found in dataset of a specified issue type.
+
+        Parameters
+        ----------
+        issue_name :
+            Name of the issue type to focus on.
+
+        Returns
+        -------
+        summary :
+            Summary of issues for a given issue type.
+        """
+        if self.issue_summary.empty:
+            raise ValueError(
+                "No issues found in the dataset. "
+                "Call `find_issues` before calling `get_summary`."
+            )
+
+        if issue_name is None:
+            return self.issue_summary
+
+        row_mask = self.issue_summary["issue_type"] == issue_name
+        if not any(row_mask):
+            raise ValueError(f"Issue type {issue_name} not found in the summary.")
+        return self.issue_summary[row_mask].reset_index(drop=True)
+
+    def _add_issue_summary_to_report(self, summary: pd.DataFrame) -> str:
+        return (
+            "Here is a summary of the different kinds of issues found in the data:\n\n"
+            + summary.to_string(index=False)
+            + "\n\n"
+            + "(Note: A lower score indicates a more severe issue across all examples in the dataset.)\n\n\n"
+        )
+
+    def find_issues(
```

Comment:
@elisno verify my new arg ordering is ok. I think the data-related args need to come first, before the issue_types arg (which is not related to the specific dataset)